src/Shikimori/Provider.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace SocialiteProviders\Shikimori;
+
+use GuzzleHttp\RequestOptions;
+use Illuminate\Support\Arr;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'SHIKIMORI';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'user_rates'
+    ];
+
+    protected $scopeSeparator = ' ';
+
+    protected function getShikimoriUrl()
+    {
+        return $this->getConfig('base_url');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getShikimoriUrl().'/oauth/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getShikimoriUrl().'/oauth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getShikimoriUrl().'/api/users/whoami', [
+            RequestOptions::HEADERS => [
+                'Authorization' => 'Bearer '.$token,
+            ],
+        ]);
+
+        return json_decode((string) $response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['id'],
+            'nickname' => $user['nickname'],
+            'name'     => $user['name'],
+            'sex'      => $user['sex'],
```

Comment:
If shikimori.one service is still being worked on
This field could also be renamed to `gender`